dask_jobqueue/core.py

```diff
@@ -407,5 +409,16 @@ def _del_pending_jobs(self):
         return jobs
 
     def _job_id_from_submit_output(self, out):
-        raise NotImplementedError('_job_id_from_submit_output must be implemented when JobQueueCluster is '
-                                  'inherited. It should convert the stdout from submit_command to the job id')
+        msg = ('Could not parse job id from submission command output. Job id '
+               'regexp is {}, submission command output is: {}'.format(
+                    self.job_id_regexp, out))
+
+        match = re.search(self.job_id_regexp, out)
+        if match is None:
+            raise ValueError(msg)
+
+        job_id = match.group('job_id')
```

Comment:
Can you use `job_id = match.groupdict().get('job_id')` to have a better error as I mentioned (probably folded away in an outdated diff).

Thinking about this it would be great to have a better error in the case when there is a match but no `job_id` named group, i.e. something like:

You need to use a `job_id` named group in your regexp, e.g. '(?P<job_id>\d+)'. Your regexp was: {}